spec/message.abnf

```diff
@@ -9,8 +9,9 @@ selectors = match 1*([s] expression)
 variant = when 1*(s key) [s] pattern
 key = nmtoken / literal / "*"
 
-expression = "{" [s] (((literal / variable) [s annotation]) / annotation) [s] "}"
+expression = "{" [s] (((literal / variable) [s annotation]) / annotation / reserved) [s] "}"
 annotation = function *(s option)
```

Comment:
Just had a thought: Should we allow for reserved operators to accept a positional argument? That could be achieved with this change:

```suggestion
expression = "{" [s] (((literal / variable) [s annotation]) / annotation) [s] "}"
annotation = (function *(s option)) / reserved
```

Reply:
We probably should do this. One callout: `reserved` is _space significant_ and has no ending marker (other than the closing `}` of the `expression` itself). An implication of this for parsers is that one can't identify a placeholder substring and `trim()` the contents before parsing. This needs to be called out in the spec.